File: Samples/Java/DecodeImage.java
/*
  DecodeImage
  Reads image file jsig.png to report capture details
  (an alternative signature image filename can be supplied as an argument)
 
  Copyright (c) 2015 Wacom GmbH. All rights reserved.
   
*/

import com.florentis.signature.SigCtl;
import com.florentis.signature.SigObj;
import com.florentis.signature.DynamicCapture;
class DecodeImage {

  private static void print(String s) {
 	System.out.println(s);
  }

	public static void main(String args[]) {
    try
      {
        String filename = "sig.png";
        if( args.length > 0 )
          filename = args[0];
        print("Reading image file: " + filename);
        SigCtl sigCtl = new SigCtl();
        sigCtl.licence("<<license>>");
        SigObj sig = sigCtl.signature();
        int rc = sig.readEncodedBitmap(filename);
        if(rc == 0) {
          print("Who: \t"+ sig.who());
          print("Why: \t" + sig.why());
          print("When: \t" + sig.when(0));
          print("Additional data: \t" + sig.extraData("AdditionalData"));
        }
        else {
          print("Error reading file: " + rc);
          switch(rc) {
            case 1: print("File not found");
                    break;
            case 2: print("File is not a supported image type");
                    break;
            case 3: print("Encoded signature data not found in image");
                    break;
          }
        }
      }
      catch (Exception ex)
      {
        print("Exception: " + ex);    
      }
	}
}

File: Samples/JScript/DecodeImage.js
/*
  DecodeImage
  Reads image file sig.png to report capture details
  (an alternative  sig.png filename can be supplied as an argument)
    
  Copyright (c) 2015 Wacom GmbH. All rights reserved.
  
*/
function print( txt ) {
  WScript.Echo(txt);
}
main();
function main() {
  filename = "sig.png";
  args = WScript.Arguments;
  if(args.Count() > 0 )
    filename=args(0);
  print("Reading image file: " + filename);
  sigCtl = new ActiveXObject("Florentis.SigCtl");
  sigCtl.SetProperty("Licence", "<<license>>");
  sig = sigCtl.Signature;
  rc = sig.ReadEncodedBitmap(filename);
  if( rc == 0 ) {
    print("Who: \t"+ sig.Who);
    print("Why: \t" + sig.Why);
    print("When: \t" + sig.When);
    print("Additional data: \t" + sig.ExtraData("AdditionalData"));
    print("Additional data: \t" + sig.AdditionalData(26));
    print("Additional data: \t" + sig.AdditionalData(27));
  }
  else {
    print("Error reading file: " + rc);
    switch(rc) {
      case 1: print("File not found");
              break;
      case 2: print("File is not a supported image type");
              break;
      case 3: print("Encoded signature data not found in image");
              break;
    }
  }
}